calibr8/tests.py

```diff
@@ -689,37 +695,79 @@ def test_infer_independent(self):
 
-        # create a pymc3 model using the calibration model
-        with pymc3.Model() as pmodel:
-            x_hat = pymc3.Uniform('x_hat', lower=0, upper=10, shape=x_true.shape, transform=None)
-
+        with pm.Model():
+            x_hat = pm.Uniform("x_hat", shape=5)
             with pytest.raises(ValueError, match="`name` must be specified"):
                 cmodel.loglikelihood(x=x_hat, y=y_obs)
 
             with pytest.warns(DeprecationWarning, match="Use `name` instead"):
-                L = cmodel.loglikelihood(x=x_hat, y=y_obs, replicate_id='A01', dependent_key='A')
-                assert isinstance(L, tt.TensorVariable)
+                cmodel.loglikelihood(x=x_hat, y=y_obs, replicate_id='A01', dependent_key='A')
+        pass
+
+    @pytest.mark.skipif(not HAS_PYMC, reason='requires PyMC')
+    def test_symbolic_loglikelihood(self):
+        cmodel = _TestPolynomialModel(independent_key='S', dependent_key='A', mu_degree=1, scale_degree=1)
+        cmodel.theta_fitted = [0, 1, 0.1, 1]
+       
+        # create test data
+        x_true = numpy.array([1,2,3,4,5])
+        y_obs = cmodel.predict_dependent(x_true)[0]
+
+        x_hat = at.vector()
+        x_hat.tag.test_value = x_true
+        L = cmodel.loglikelihood(x=x_hat, y=y_obs, name='L_A01_A')
+        assert isinstance(L, at.TensorVariable)
+        assert L.ndim == 0
+
+        # compare the two loglikelihood computation methods
+        x_test = numpy.random.normal(x_true, scale=0.1)
+        actual = L.eval({
+            x_hat: x_test
+        })
+        expected = cmodel.loglikelihood(x=x_test, y=y_obs)
+        assert numpy.ndim(expected) == 0
+        assert numpy.ndim(actual) == 0
+        numpy.testing.assert_almost_equal(actual, expected, 6)
+        pass
+
+    @pytest.mark.skipif(not HAS_PYMC, reason='requires PyMC')
+    def test_symbolic_loglikelihood_in_modelcontext(self):
+        cmodel = _TestPolynomialModel(independent_key='S', dependent_key='A', mu_degree=1, scale_degree=1)
+        cmodel.theta_fitted = [0, 0.5, 0.1, 1]
+       
+        # create test data
+        x_true = numpy.array([1,2,3,4,5])
+        y_obs = cmodel.predict_dependent(x_true)[0]
 
+        # create a PyMC model using the calibration model
+        with pm.Model() as pmodel:
+            x_hat = pm.Uniform("x_hat", 0, 1, shape=x_true.shape, transform=None)
             L = cmodel.loglikelihood(x=x_hat, y=y_obs, name='L_A01_A')
-            assert isinstance(L, tt.TensorVariable)
-        
+        assert isinstance(L, at.TensorVariable)
+        assert L.ndim == 0
+
+        # PyMC v4 returns the RV, but for .eval() we need the RV-value-variable
+        if pm.__version__[0] != "3":
+            x_hat = pmodel.rvs_to_values[x_hat]
+
         # compare the two loglikelihood computation methods
         x_test = numpy.random.normal(x_true, scale=0.1)
-        actual = L.logp({
-            'x_hat': x_test
+        actual = L.eval({
```

Comment:
eval is not working for older PyMC3 versions, is it? We should update the verswion requirement in case PyMC3 is used.